<h4 align="center" style="height:1123px;font-family: 'Times New Roman'"><b>

    LAPORAN PRAKTIKUM

    ALGORITMA & PEMOGRAMAN

    “OPERATOR PADA PYTHON”

    Disusun Oleh :

    Maulani Almufida Ilmi

    2611531006

    Dosen Pengampu:

    Wahyudi.DR.. S.T.M.T

    Asisten Praktikum:

    Muhammad Galid Avero

<br>

 <img src="images/logo_unand.png" width="200">

<br>

    FAKULTAS TEKNOLOGI INFORMASI

    DEPARTEMEN INFORMATIKA

    UNIVERSITAS ANDALAS

    2026

</b></h4>
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">

<h2 align="center">KATA PENGANTAR</h2>
Puji dan syukur penulis ucapkan ke hadirat Tuhan Yang Maha Esa atas segala rahmat dan karunia-Nya, sehingga penulis dapat menyelesaikan laporan praktikum Pertemuan ke-3 pada mata kuliah Praktikum Operasi dan Python dengan baik. Laporan ini dibuat berdasarkan kegiatan praktikum yang telah dilaksanakan serta sebagai salah satu bentuk dokumentasi dari proses pembelajaran yang telah dilakukan.

Pada praktikum ini, penulis mempelajari dan memahami materi mengenai Operasi dan Python. Kegiatan praktikum diharapkan dapat membantu penulis dalam memahami cara kerja Aritmatika, Assignment, Bitwise, Lainnya, Logika, dan Perbandingan. Tidak hanya melalui pemahaman teori, tetapi juga dengan melakukan implementasi secara langsung pada sebuah real project.

Penulis menyadari bahwa dalam penyusunan laporan praktikum ini masih terdapat kekurangan dan keterbatasan. Oleh karena itu, penulis sangat mengharapkan kritik dan saran yang bersifat membangun untuk menyempurnakan laporan ini di masa yang akan datang.

Akhir kata, penulis berharap laporan ini dapat memberikan manfaat dan menambah wawasan bagi penulis maupun pembaca dalam memahami materi yang berkaitan dengan Praktikum Operasi dan Python.

<div align="right" style="margin-left:-30px;">
Padang, 2026
<br>
<br>
<br>
<br>
<br>
Maulani Almufida Ilmi
<br>
2611531006
</div>
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB I<br>PENDAHULUAN</h1>

## 1.1 Latar Belakang
Python merupakan bahasa pemrograman tingkat tinggi yang mudah dipelajari, memiliki struktur data yang efisien, serta mendukung pendekatan pemrograman berorientasi objek secara sederhana namun efektif. Sintaksnya yang elegan dikombinasikan dengan sistem penulisan tipe data dinamis dan sifatnya sebagai bahasa interpreter menjadikan Python ideal untuk scripting maupun pengembangan aplikasi secara cepat di berbagai platform. Karakteristik inilah yang membuat Python banyak diminati, baik di lingkungan akademik maupun industri, sebagai bahasa pemrograman untuk berbagai keperluan. 

Dalam mempelajari Python, salah satu fondasi utama yang harus dikuasai adalah operator. Operator merupakan simbol khusus yang melakukan suatu operasi pada operand dan mengembalikan suatu hasil, misalnya operasi penjumlahan antara dua nilai. Operator digunakan untuk melakukan berbagai aksi terhadap variabel maupun nilai numerik dan objek lain yang disebut operand. Python mengelompokkan operator ke dalam beberapa kategori, yaitu operator aritmatika, assignment, bitwise, perbandingan, logika, identitas, dan keanggotaan. Pemahaman terhadap seluruh kategori operator ini menjadi dasar penting sebelum mempelajari struktur kontrol dan logika program yang lebih kompleks, sehingga praktikum ini dilakukan untuk mempelajari dan mengimplementasikan langsung setiap jenis operator tersebut dalam bahasa Python.

## 1.2 Tujuan
1. Memahami konsep dasar operator dalam bahasa pemrograman Python.
2. Mengetahui jenis-jenis operator dalam Python (aritmatika, assignment, bitwise, logika, perbandingan, dan operator lainnya).
3. Mampu mengimplementasikan setiap jenis operator ke dalam program Python.
4. Melatih logika berpikir dalam menyelesaikan permasalahan menggunakan operator yang tepat.
## 1.3 Manfaat
1. Menambah pemahaman mengenai fungsi dan penggunaan operator dalam Python.
2. Meningkatkan kemampuan praktik dalam menulis dan menjalankan program.
3. Melatih ketelitian dan kedisiplinan dalam menyusun laporan praktikum secara sistematis.
4. Menjadi bekal dasar untuk mempelajari konsep pemrograman Python yang lebih kompleks.
</div>

---
<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB II<br>PEMBAHASAN</h1>

## 2.1 Dasar Teori
Dalam bahasa pemrograman Python, operator adalah simbol khusus yang digunakan untuk melakukan suatu operasi atau manipulasi terhadap satu, dua, atau lebih nilai (disebut operand). Operator merupakan salah satu komponen fundamental dalam pemrograman karena hampir setiap program membutuhkan proses perhitungan, pengambilan keputusan, maupun manipulasi data yang melibatkan penggunaan operator.

Secara umum, operator dalam Python berfungsi untuk menghasilkan suatu nilai baru berdasarkan operand yang diproses. Sebagai contoh, operator dapat digunakan untuk menjumlahkan dua angka, membandingkan dua nilai untuk menentukan mana yang lebih besar, atau menggabungkan beberapa kondisi logika untuk pengambilan keputusan dalam program.

Python menyediakan berbagai jenis operator yang dikelompokkan berdasarkan fungsinya, antara lain:

### 1. Operator Aritmatika
Operator aritmatika digunakan untuk melakukan operasi matematis dasar antara dua nilai atau lebih. Python menyediakan beberapa operator aritmatika, yaitu:
- `+` (penjumlahan)
- `-` (pengurangan)
- `*` (perkalian)
- `/` (pembagian, menghasilkan bilangan desimal/float)
- `//` (pembagian bulat/floor division, hasil dibulatkan ke bawah)
- `%` (modulus, menghasilkan sisa pembagian)
- `**` (pemangkatan/eksponen)

### 2. Operator Assignment (Penugasan)
Operator penugasan digunakan untuk memberikan nilai ke suatu variabel. Selain operator `=` yang sederhana, Python juga menyediakan *compound assignment* yang menggabungkan operasi aritmatika dengan penugasan, seperti `+=`, `-=`, `*=`, `/=`, `//=`, `%=`, dan `**=`. Operator ini berguna untuk memperbarui nilai variabel secara ringkas tanpa menulis ulang nama variabelnya.

### 3. Operator Bitwise
Operator bitwise bekerja pada representasi biner (bit) dari suatu bilangan bulat. Operator ini meliputi:
- `&` (AND) — menghasilkan 1 jika kedua bit bernilai 1
- `|` (OR) — menghasilkan 1 jika salah satu bit bernilai 1
- `^` (XOR) — menghasilkan 1 jika kedua bit berbeda
- `~` (NOT) — membalik seluruh bit (komplemen)
- `<<` (shift left) — menggeser bit ke kiri
- `>>` (shift right) — menggeser bit ke kanan

Operator ini sering digunakan dalam pemrograman tingkat rendah, manipulasi flag, dan optimasi.

### 4. Operator Logika (Logical Operator)
Operator logika digunakan untuk menggabungkan atau mengevaluasi ekspresi kondisional yang menghasilkan nilai boolean (`True`/`False`). Terdiri dari:
- `and` — bernilai True jika kedua kondisi benar
- `or` — bernilai True jika salah satu kondisi benar
- `not` — membalik nilai boolean

Operator ini banyak digunakan dalam struktur kontrol seperti percabangan (`if`) dan perulangan.

### 5. Operator Perbandingan (Comparison Operator)
Operator perbandingan digunakan untuk membandingkan dua nilai, dan hasilnya selalu berupa nilai boolean. Operatornya meliputi:
- `==` (sama dengan)
- `!=` (tidak sama dengan)
- `>` (lebih besar dari)
- `<` (lebih kecil dari)
- `>=` (lebih besar sama dengan)
- `<=` (lebih kecil sama dengan)

### 6. Operator Lainnya
Kategori ini biasanya mencakup operator khusus Python seperti:
- **Operator Identitas** (`is`, `is not`) — membandingkan apakah dua variabel merujuk ke objek yang sama di memori
- **Operator Keanggotaan** (`in`, `not in`) — memeriksa apakah suatu nilai terdapat dalam suatu urutan (list, string, tuple, dll.)


## 2.2 Kode Program

### 2.2.1 Program aritmatika_2611531006
</div>

In [1]:
# Buat program untuk operator aritmatika dalam Python

angka1_1006 = int(input("Input angka-1: "))
angka2_1006 = int(input("Input angka-2: "))

# Penjumlahan
hasil_1006 = angka1_1006 + angka2_1006
print("\nOperator Penjumlahan")
print("Hasil =",hasil_1006)

# Pengurangan
hasil_1006 = angka1_1006 - angka2_1006
print("\nOperator Pengurangan")
print("Hasil =",hasil_1006)

# Perkalian
hasil_1006 = angka1_1006 * angka2_1006
print("\nOperator Perkalian")
print("Hasil =",hasil_1006)

# Pembagian, pembagian bulat, dan sisa bagi
if angka2_1006 != 0:
    hasil_1006 = angka1_1006 / angka2_1006
    print("\nOperator Pembagian")
    print("Hasil =",hasil_1006)

    hasil_1006 = angka1_1006 // angka2_1006
    print("\nOperator Pembagian Bulat")
    print("Hasil =",hasil_1006)

    hasil_1006 = angka1_1006 % angka2_1006
    print("\nOperator Sisa Bagi")
    print("Hasil =",hasil_1006)
else:
    print("Angka kedua tidak boleh bernilai 0.")

# Pangkat
hasil_1006 = angka1_1006 ** angka2_1006
print("\nOperator Pangkat")
print("Hasil =",hasil_1006)


Operator Penjumlahan
Hasil = 30

Operator Pengurangan
Hasil = -10

Operator Perkalian
Hasil = 200

Operator Pembagian
Hasil = 0.5

Operator Pembagian Bulat
Hasil = 0

Operator Sisa Bagi
Hasil = 10

Operator Pangkat
Hasil = 100000000000000000000


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. **Input Data**
   ```python
   angka1_1006 = int(input("Input angka-1: "))
   angka2_1006 = int(input("Input angka-2: "))
   ```
   Pengguna memasukkan dua nilai angka melalui `input()`, yang dikonversi ke tipe integer menggunakan `int()`.

2. **Operator Penjumlahan, Pengurangan, dan Perkalian**
   ```python
   hasil_1006 = angka1_1006 + angka2_1006
   hasil_1006 = angka1_1006 - angka2_1006
   hasil_1006 = angka1_1006 * angka2_1006
   ```
   Ketiga operator (`+`, `-`, `*`) langsung dijalankan tanpa syarat khusus, karena selalu dapat dilakukan terhadap dua angka berapa pun.

3. **Operator Pembagian, Pembagian Bulat, dan Modulus**
   ```python
   if angka2_1006 != 0:
       hasil_1006 = angka1_1006 / angka2_1006
       hasil_1006 = angka1_1006 // angka2_1006
       hasil_1006 = angka1_1006 % angka2_1006
   else:
       print("Angka kedua tidak boleh bernilai 0.")
   ```
   Digunakan struktur `if-else` karena operasi pembagian (`/`), pembagian bulat (`//`), dan modulus (`%`) akan menyebabkan `ZeroDivisionError` jika `angka2_1006` bernilai 0.

4. **Operator Pangkat**
   ```python
   hasil_1006 = angka1_1006 ** angka2_1006
   ```
   Operator `**` menghitung pemangkatan `angka1_1006` terhadap `angka2_1006`, dijalankan di luar `if-else` karena tidak memiliki batasan seperti pembagian.

### 2.2.2 Program assigment_2611531006
</div>

In [2]:
# Program operator assigment dalam Python

angka1_1006 = int(input("Input angka-1: "))
angka2_1006 = int(input("Input angka-2: "))

print("\nNilai awal angka1 =",angka1_1006)
print("Nilai angka2 =",angka2_1006)

# Assigment biasa
hasil_1006 = angka1_1006 
print("\nAssigment biasa (=)")
print("Hasil =",hasil_1006)

# Assigment penambahan
hasil_1006 = angka1_1006 
hasil_1006 += angka2_1006 
print("\nAssigment penambahan (+=)")
print("Hasil =",hasil_1006)

# Assigment pengurangan
hasil_1006 = angka1_1006 
hasil_1006 -= angka2_1006 
print("\nAssigment pengurangan (-=)")
print("Hasil =",hasil_1006)

# Assigment perkalian
hasil_1006 = angka1_1006 
hasil_1006 *= angka2_1006
print("\nAssigment perkalian (*=)")
print("Hasil =",hasil_1006)

# Assigment pembagian, pembagian bulat, dan sisa bagi
if angka2_1006 != 0:
    hasil_1006 = angka1_1006 
    hasil_1006 /= angka2_1006
    print("\nAssigment pembagian (/=)")
    print("Hasil =",hasil_1006)
    # Operator tambahan
    hasil_1006 = angka1_1006 
    hasil_1006 //= angka2_1006
    print("\nAssigment pembagian bulat (//=)")
    print("Hasil =",hasil_1006)
    hasil_1006 = angka1_1006 
    hasil_1006 %= angka2_1006
    print("\nOperator sisa bagi (%=)")
    print("Hasil =",hasil_1006)
else:
    print("\nPembagian tidak dapat dilakukan.")
    print("Angka kedua tidak boleh bernilai 0.")

# Operator tambahan: assigment perpangkatan
hasil_1006 = angka1_1006 
hasil_1006 **= angka2_1006
print("\nOperator perpangkatan (**=)")
print("Hasil =",hasil_1006)


Nilai awal angka1 = 10
Nilai angka2 = 20

Assigment biasa (=)
Hasil = 10

Assigment penambahan (+=)
Hasil = 30

Assigment pengurangan (-=)
Hasil = -10

Assigment perkalian (*=)
Hasil = 200

Assigment pembagian (/=)
Hasil = 0.5

Assigment pembagian bulat (//=)
Hasil = 0

Operator sisa bagi (%=)
Hasil = 10

Operator perpangkatan (**=)
Hasil = 100000000000000000000


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. **Input Data**
   ```python
   angka1_1006 = int(input("Input angka-1: "))
   angka2_1006 = int(input("Input angka-2: "))
   ```
   Pengguna memasukkan dua nilai angka melalui `input()`, yang dikonversi ke tipe integer menggunakan `int()`.

2. **Assignment Biasa (=)**
   ```python
   hasil_1006 = angka1_1006
   ```
   Operator `=` digunakan untuk memberikan nilai `angka1_1006` langsung ke variabel `hasil_1006` tanpa operasi tambahan.

3. **Assignment Penambahan, Pengurangan, dan Perkalian (+=, -=, *=)**
   ```python
   hasil_1006 += angka2_1006
   hasil_1006 -= angka2_1006
   hasil_1006 *= angka2_1006
   ```
   Ketiga operator ini merupakan bentuk ringkas dari operasi aritmatika yang digabung dengan assignment. Misalnya, `hasil_1006 += angka2_1006` setara dengan `hasil_1006 = hasil_1006 + angka2_1006`.

4. **Assignment Pembagian, Pembagian Bulat, dan Modulus (/=, //=, %=)**
   ```python
   if angka2_1006 != 0:
       hasil_1006 /= angka2_1006
       hasil_1006 //= angka2_1006
       hasil_1006 %= angka2_1006
   else:
       print("Pembagian tidak dapat dilakukan.")
   ```
   Digunakan struktur `if-else` karena operator `/=`, `//=`, dan `%=` akan menyebabkan `ZeroDivisionError` jika `angka2_1006` bernilai 0.

5. **Assignment Perpangkatan (**=)**
   ```python
   hasil_1006 **= angka2_1006
   ```
   Operator `**=` menghitung pemangkatan `hasil_1006` terhadap `angka2_1006`, dijalankan di luar `if-else` karena tidak memiliki batasan seperti pembagian.

### 2.2.3 Program perbandingan_2611531006
</div>

In [ ]:
# Program operator perbandingan dalam Python

angka1_1006 = int(input("Input angka-1: "))
angka2_1006 = int(input("Input angka-2: "))

# Lebih besar dari
hasil_1006 = angka1_1006 > angka2_1006
print("\nOperator lebih besar dari")
print("angka1 > angka2 =",hasil_1006)

# Lebih kecil dari
hasil_1006 = angka1_1006 < angka2_1006
print("\nOperator lebih kecil dari")
print("angka1 < angka2 =",hasil_1006)

# Lebih besar dari atau sama dengan
hasil_1006 = angka1_1006 >= angka2_1006
print("\nOperator lebih besar dari atau sama dengan")
print("angka1 >= angka2 =",hasil_1006)

# Lebih kecil dari atau sama dengan
hasil_1006 = angka1_1006 <= angka2_1006
print("\nOperator lebih kecil dari atau sama dengan")
print("angka1 <= angka2 =",hasil_1006)

# Sama dengan
hasil_1006 = angka1_1006 == angka2_1006
print("\nOperator sama dengan")
print("angka1 == angka2 =",hasil_1006)

# Tidak sama dengan
hasil_1006 = angka1_1006 != angka2_1006
print("\nOperator tidak sama dengan")
print("angka1 != angka2 =",hasil_1006)

# Tambahan: perbandingan berantai dalam Python
hasil_1006 = 0 < angka1_1006 < 100
print("\nPerbandingan berantai")
print("0 < angka1_1006 < 100 =",hasil_1006)

hasil_1006 = 0 < angka2_1006 < 100
print("0 < angka2_1006 < 100 =",hasil_1006)


Operator lebih besar dari
angka1 > angka2 = False

Operator lebih kecil dari
angka1 < angka2 = True

Operator lebih besar dari atau sama dengan
angka1 >= angka2 = False

Operator lebih kecil dari atau sama dengan
angka1 <= angka2 = True

Operator sama dengan
angka1 == angka2 = False

Operator tidak sama dengan
angka1 != angka2 = True

Perbandingan berantai
0 < angka1_1021 < 100 = True
0 < angka2_1021 < 100 = True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Input Data
```python
angka1_1006 = int(input("Input angka-1: "))
angka2_1006 = int(input("Input angka-2: "))
```
Meminta dua angka dari pengguna. Hasil `input()` berupa string, jadi dibungkus `int()` supaya bisa dibandingkan secara numerik.

2. Enam Operator Perbandingan Dasar
Setiap blok mengikuti pola yang sama: hitung hasil perbandingan, lalu cetak dengan judul dan hasilnya.

| Operator | Arti | Kode |
|---|---|---|
| `>` | lebih besar dari | `angka1_1006 > angka2_1006` |
| `<` | lebih kecil dari | `angka1_1006 < angka2_1006` |
| `>=` | lebih besar atau sama dengan | `angka1_1006 >= angka2_1006` |
| `<=` | lebih kecil atau sama dengan | `angka1_1006 <= angka2_1006` |
| `==` | sama dengan (bukan assignment!) | `angka1_1006 == angka2_1006` |
| `!=` | tidak sama dengan | `angka1_1006 != angka2_1006` |

Variabel `hasil_1006` dipakai berulang-ulang (di-*overwrite* setiap blok), lalu langsung dicetak sebelum pindah ke operator berikutnya.

Catatan kecil: `print("\nOperator lebih besar dari")` — `\n` di awal string membuat baris kosong sebelum judul, supaya outputnya lebih rapi dan terpisah antar blok.

3. Perbandingan Berantai (Chained Comparison)
```python
hasil_1006 = 0 < angka1_1006 < 100
hasil_1006 = 0 < angka2_1006 < 100
```
Fitur khas Python ini setara dengan `(0 < x) and (x < 100)` — mengecek apakah angka berada di rentang 1–99. Di sini dilakukan dua kali: sekali untuk `angka1_1006`, sekali untuk `angka2_1006`.

### 2.2.4 Program logika_2611531006
</div>

In [4]:
# Program operator logika dalam Python
# Memasukan nilai boolean
# Input tidak peka terhadap huruf besar dan kecil
a1_1006 = input("Input nilai boolean-1 (true/false): ").strip().lower() == "true"
a2_1006 = input("Input nilai boolean-2 (true/false): ").strip().lower() == "true"

print("\nA1 = ",a1_1006)
print("A2 = ",a2_1006)

# Konjungsi: bernilai True jika keduanya True
hasil_1006 = a1_1006 and a2_1006
print("\nKonjungsi (AND)")
print("A1 and A2 =",hasil_1006)

# Disjungsi: bernilai True jika salah satunya False
hasil_1006 = a1_1006 or a2_1006
print("\nDisjungsi (OR)")
print("A1 or A2 =",hasil_1006)

# Negasi A1: membalik nilai A1
hasil_1006 = not a1_1006
print("\nNegasi A1(NOT)")
print("not A1",hasil_1006)

# Negasi A2: membalik nilai A2
hasil_1006 = not a2_1006
print("\nNegasi A2(NOT)")
print("not A2",hasil_1006)

# XOR: bernilai True jika kedua nilai berbeda
hasil_1006 = a1_1006 != a2_1006
print("\nDisjungsi Ekslusif (XOR)")
print("A1 XOR A2",hasil_1006)


A1 =  True
A2 =  False

Konjungsi (AND)
A1 and A2 = False

Disjungsi (OR)
A1 or A2 = True

Negasi A1(NOT)
not A1 False

Negasi A2(NOT)
not A2 True

Disjungsi Ekslusif (XOR)
A1 XOR A2 True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:

**1. Input nilai boolean**
```python
a1_1006 = input("Input nilai boolean-1 (true/false): ").strip().lower() == "true"
a2_1006 = input("Input nilai boolean-2 (true/false): ").strip().lower() == "true"
```
Pengguna mengetik teks `true`/`false`. `.strip()` membuang spasi di awal/akhir, `.lower()` mengubah ke huruf kecil sehingga input seperti `TRUE`, `True`, `true` semuanya dianggap sama. Hasil perbandingan `== "true"` menghasilkan nilai boolean `True` atau `False`, disimpan ke `a1_1006` dan `a2_1006`.

**2. Menampilkan nilai input**
```python
print("\nA1 = ",a1_1006)
print("A2 = ",a2_1006)
```
Menampilkan kembali nilai boolean yang dimasukkan, untuk verifikasi.

**3. Konjungsi (AND)**
```python
hasil_1006 = a1_1006 and a2_1006
```
Bernilai `True` hanya jika **kedua** nilai `True`. Jika salah satu `False`, hasilnya `False`.

**4. Disjungsi (OR)**
```python
hasil_1006 = a1_1006 or a2_1006
```
Bernilai `True` jika **salah satu** nilai `True`. Hanya `False` jika keduanya `False`.

**5. Negasi (NOT)**
```python
hasil_1006 = not a1_1006
...
hasil_1006 = not a2_1006
```
Membalik nilai boolean: `True` menjadi `False` dan sebaliknya. Dilakukan terpisah untuk A1 dan A2.

**6. Disjungsi Eksklusif (XOR)**
```python
hasil_1006 = a1_1006 != a2_1006
```
Python tidak punya operator XOR khusus untuk boolean, sehingga dipakai `!=` (tidak sama dengan). Hasilnya `True` jika kedua nilai **berbeda** (satu `True`, satu `False`), dan `False` jika kedua nilai sama.

**Tabel kebenaran ringkas:**

| A1 | A2 | AND | OR | XOR |
|---|---|---|---|---|
| True | True | True | True | False |
| True | False | False | True | True |
| False | True | False | True | True |
| False | False | False | False | False |

Mau saya buatkan juga versi file (Word/PDF) dari penjelasan ini?
### 2.2.5 Program lainnya_2611531006
</div>

In [ ]:
# Program keanggotaan dan identitas dalam python

print("======================================")
print("1. OPERATOR KEANGGOTAAN")
print("======================================")

# Input beberapa data yang dipisahkan dengan koma
input_data_1006 = input("Masukkan beberapa angka, pisahkan dengan koma: ")

# Mengubah input menjadi list integer
data_1006 = [int(angka.strip()) for angka in input_data_1006.split(',')]

nilai_dicari_1006 = int(input("Masukkan angka yang ingin dicari: "))

# Operator in
hasil_1006 = nilai_dicari_1006 in data_1006
print("\nOperator keanggotaan IN")
print(nilai_dicari_1006,"in",data_1006,"=",hasil_1006)

# Operator not in
hasil_1006 = nilai_dicari_1006 not in data_1006
print("\nOperator keanggotaan NOT IN")
print(nilai_dicari_1006,"not in",data_1006,"=",hasil_1006)

print("======================================")
print("2. OPERATOR IDENTIAS")
print("======================================")

# objek1 menggunakan list dari input perngguna
object1_1006 = data_1006

# objek2 merujuk pada objek yang sama dengan objek1
object2_1006 = object1_1006

# objek3 memiliki isi sama, tetapi merupakan objek baru
object3_1006 = data_1006.copy()

print("object1 =",object1_1006)
print("object2 =",object2_1006)
print("object3 =",object3_1006)

# Operator is
hasil_1006 = object1_1006 is object2_1006
print("\nOperator identitas IS")
print("objek1 is objek2 =",hasil_1006)

# Operator is not
hasil_1006 = object1_1006 is not object2_1006
print("\nOperator identitas IS NOT")
print("objek1 is not objek2 =",hasil_1006)

# Membandingkan identitas dan nilai
print("\nPerbandingan identitas dan nilai")
print("objek1 is objek3 =",object1_1006 is object3_1006)
print("objek1 == objek3 =",object1_1006 == object3_1006)

1. OPERATOR KEANGGOTAAN

Operator keanggotaan IN
20 in [10, 20, 30, 40, 50] = True

Operator keanggotaan NOT IN
20 not in [10, 20, 30, 40, 50] = False
2. OPERATOR IDENTIAS
object1 = [10, 20, 30, 40, 50]
object2 = [10, 20, 30, 40, 50]
object3 = [10, 20, 30, 40, 50]

Operator identitas IS
objek1 is objek2 = True

Operator identitas IS NOT
objek1 is not objek2 = False

Perbandingan identitas dan nilai
objek1 is objek3 = False
objek1 == objek3 = True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
1. Operator Keanggotaan

**Input data**
```python
input_data_1006 = input("Masukkan beberapa angka, pisahkan dengan koma: ")
data_1006 = [int(angka.strip()) for angka in input_data_1006.split(',')]
nilai_dicari_1006 = int(input("Masukkan angka yang ingin dicari: "))
```
Pengguna memasukkan beberapa angka dipisah koma (misal `1, 2, 3`). `.split(',')` memecah teks itu menjadi list string berdasarkan koma, `.strip()` membuang spasi di setiap elemen, lalu `int(...)` mengubah tiap elemen menjadi angka. List comprehension `[int(angka.strip()) for angka in ...]` melakukan semua ini sekaligus, menghasilkan `data_1006` berupa list integer, misalnya `[1, 2, 3]`.

Operator `in`
```python
hasil_1006 = nilai_dicari_1006 in data_1006
```
Bernilai `True` jika `nilai_dicari_1006` **ditemukan** di dalam list `data_1006`, dan `False` jika tidak ada.

Operator `not in`
```python
hasil_1006 = nilai_dicari_1006 not in data_1006
```
Kebalikan dari `in` — bernilai `True` jika angka tersebut **tidak ditemukan** dalam list.

2. Operator Identitas

**Membuat tiga referensi objek**
```python
object1_1006 = data_1006
object2_1006 = object1_1006
object3_1006 = data_1006.copy()
```
- `object1_1006` menunjuk ke list `data_1006` yang sama persis di memori.
- `object2_1006 = object1_1006` membuat `object2_1006` menunjuk ke **objek yang sama** dengan `object1_1006` (bukan salinan) — keduanya adalah nama berbeda untuk list yang sama di memori.
- `object3_1006 = data_1006.copy()` membuat **salinan baru** (objek berbeda di memori) yang isinya sama dengan `data_1006`.

Operator `is`
```python
hasil_1006 = object1_1006 is object2_1006
```
Bernilai `True` jika kedua variabel menunjuk ke **objek yang sama persis di memori** (bukan sekadar nilainya sama). Karena `object2_1006` dibuat dengan `= object1_1006`, keduanya menunjuk ke objek yang sama, sehingga hasilnya `True`.

Operator `is not`
```python
hasil_1006 = object1_1006 is not object2_1006
```
Kebalikan dari `is` — bernilai `True` jika keduanya **bukan objek yang sama**. Karena `object1_1006` dan `object2_1006` adalah objek yang sama, hasilnya `False`.

Perbandingan identitas vs nilai
```python
print("objek1 is objek3 =",object1_1006 is object3_1006)
print("objek1 == objek3 =",object1_1006 == object3_1006)
```
Ini bagian kunci untuk memahami perbedaan `is` dan `==`:
- `object1_1006 is object3_1006` → `False`, karena `object3_1006` adalah **objek baru** hasil `.copy()`, meskipun isinya identik, lokasinya di memori berbeda.
- `object1_1006 == object3_1006` → `True`, karena `==` membandingkan **isi/nilai** list, bukan lokasi memorinya, dan isi kedua list sama.

### 2.2.6 Program bitwise_2611531006
</div>

In [ ]:
# Buat file dengan nama bitwise_NIM.py
# Nama variabel ditambah 4 digit nim terakhir contoh: angka1_1234
# Program ini menggunakan fungsi input()

print("======================================")
print("3. OPERATOR BITWISE")
print("======================================")

angka1_1006 = int(input("Masukkan angka bitwise-1: "))
angka2_1006 = int(input("Masukkan angka bitwise-2: "))

print("\nAngka dalam bentuk desimal dan biner")
print("angka1 =",angka1_1006,"| biner =",bin(angka1_1006))
print("angka1 =",angka2_1006,"| biner =",bin(angka2_1006))

# Bitwise AND
hasil_1006 = angka1_1006 & angka2_1006
print("\nBitwise AND (&)")
print(angka1_1006,"&",angka2_1006,hasil_1006)
print("Biner hasil =",bin(hasil_1006))
print("Biner hasil (8 bit) =",format(hasil_1006,"08b"))

# Bitwise OR
hasil_1006 = angka1_1006 | angka2_1006
print("\nBitwise OR (|)")
print(angka1_1006,"|",angka2_1006,hasil_1006)
print("Biner hasil =",bin(hasil_1006))
print("Biner hasil (8 bit) =",format(hasil_1006,"08b"))

# Bitwise XOR
hasil_1006 = angka1_1006 ^ angka2_1006
print("\nBitwise XOR (^)")
print(angka1_1006,"^",angka2_1006,hasil_1006)
print("Biner hasil =",bin(hasil_1006))
print("Biner hasil (8 bit) =",format(hasil_1006,"08b"))

# Bitwise NOT
hasil_1006 = ~angka1_1006
print("\nBitwise NOT (~)")
print(angka1_1006,"~",angka2_1006,hasil_1006)
print("Biner hasil =",bin(hasil_1006))
print("Biner hasil (8 bit) =",format(hasil_1006,"08b"))

# Bitwise geser kiri
jumlah_geser_1006 = int(input("\nMasukkan jumlah pergeseran bit: "))

hasil_1006 = angka1_1006 << jumlah_geser_1006
print("\nBitwise geser kiri (<<)")
print(angka1_1006,"<<",jumlah_geser_1006,"=",hasil_1006)
print("Biner hasil =",bin(hasil_1006))
print("Biner hasil (8 bit) =",format(hasil_1006,"08b"))

# Bitwise geser kanan
hasil_1006 = angka1_1006 >> jumlah_geser_1006
print("\nBitwise geser kanan (>>)")
print(angka1_1006,">>",jumlah_geser_1006,"=",hasil_1006)
print("Biner hasil =",bin(hasil_1006))
print("Biner hasil (8 bit) =",format(hasil_1006,"08b"))

3. OPERATOR BITWISE

Angka dalam bentuk desimal dan biner
angka1 = 10 | biner = 0b1010
angka1 = 20 | biner = 0b10100

Bitwise AND (&)
10 & 20 0
Biner hasil = 0b0
Biner hasil (8 bit) = 00000000

Bitwise OR (|)
10 | 20 30
Biner hasil = 0b11110
Biner hasil (8 bit) = 00011110

Bitwise XOR (^)
10 ^ 20 30
Biner hasil = 0b11110
Biner hasil (8 bit) = 00011110

Bitwise NOT (~)
10 ~ 20 -11
Biner hasil = -0b1011
Biner hasil (8 bit) = -0001011

Bitwise geser kiri (<<)
10 << 1 = 20
Biner hasil = 0b10100
Biner hasil (8 bit) = 00010100

Bitwise geser kanan (>>)
10 >> 1 = 5
Biner hasil = 0b101
Biner hasil (8 bit) = 00000101


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:
Berikut penjelasan yang sama dengan digit diganti dari **1021** menjadi **1006**.

1. Judul bagian
```python
print("======================================")
print("3. OPERATOR BITWISE")
print("======================================")
```
Menampilkan judul agar output program terlihat terstruktur per bagian.

2. Input dua angka
```python
angka1_1006 = int(input("Masukkan angka bitwise-1: "))
angka2_1006 = int(input("Masukkan angka bitwise-2: "))
```
`input()` menerima masukan berupa teks, lalu `int()` mengonversinya menjadi bilangan bulat (integer) agar bisa dioperasikan secara bitwise.

3. Menampilkan bentuk desimal dan biner
```python
print("\nAngka dalam bentuk desimal dan biner")
print("angka1 =",angka1_1006,"| biner =",bin(angka1_1006))
print("angka1 =",angka2_1006,"| biner =",bin(angka2_1006))
```
`bin()` mengubah bilangan integer menjadi representasi biner (string berawalan `0b`), agar pengguna bisa melihat bentuk bit dari kedua angka sebelum dioperasikan.

4. Operasi AND (`&`)
```python
hasil_1006 = angka1_1006 & angka2_1006
print(angka1_1006,"&",angka2_1006,hasil_1006)
print("Biner hasil =",bin(hasil_1006))
print("Biner hasil (8 bit) =",format(hasil_1006,"08b"))
```
Operator `&` membandingkan bit-per-bit kedua angka; suatu bit hasil bernilai `1` hanya jika **kedua** bit yang dibandingkan bernilai `1`. `format(hasil_1006, "08b")` menampilkan hasil dalam biner dengan panjang tetap **minimal 8 digit** (diisi nol di depan jika kurang), berbeda dari `bin()` yang panjangnya mengikuti nilai aslinya.

5. Operasi OR (`|`)
```python
hasil_1006 = angka1_1006 | angka2_1006
print(angka1_1006,"|",angka2_1006,hasil_1006)
print("Biner hasil =",bin(hasil_1006))
print("Biner hasil (8 bit) =",format(hasil_1006,"08b"))
```
Operator `|` menghasilkan bit `1` jika **salah satu atau kedua** bit bernilai `1`. Hasil hanya `0` jika kedua bit sama-sama `0`.

6. Operasi XOR (`^`)
```python
hasil_1006 = angka1_1006 ^ angka2_1006
```
Operator `^` menghasilkan bit `1` jika kedua bit **berbeda**, dan `0` jika kedua bit **sama**:

| Bit A | Bit B | A ^ B |
|---|---|---|
| 0 | 0 | 0 |
| 0 | 1 | 1 |
| 1 | 0 | 1 |
| 1 | 1 | 0 |

Baris `print()` berikutnya menampilkan hasil XOR dalam bentuk desimal dan biner, sama seperti bagian AND/OR.

7. Operasi NOT (`~`)
```python
hasil_1006 = ~angka1_1006
print(angka1_1006,"~",angka2_1006,hasil_1006)
```
Operator `~` membalik seluruh bit dari `angka1_1006` (bit `1` menjadi `0`, bit `0` menjadi `1`) — ini disebut *komplemen satu* (one's complement). Secara matematis, untuk bilangan bulat di Python, `~x` sama dengan `-x - 1`, sehingga hasilnya selalu bernilai negatif dari angka positif.

```python
print("~", angka1_1006, "=", hasil_1006)
```

8. Input jumlah pergeseran bit
```python
jumlah_geser_1006 = int(input("\nMasukkan jumlah pergeseran bit: "))
```
Meminta pengguna menentukan berapa posisi bit yang akan digeser, dikonversi menjadi integer.

9. Pergeseran bit ke kiri (`<<`)
```python
hasil_1006 = angka1_1006 << jumlah_geser_1006
print("\nBitwise geser kiri (<<)")
print(angka1_1006,"<<",jumlah_geser_1006,"=",hasil_1006)
print("Biner hasil =",bin(hasil_1006))
print("Biner hasil (8 bit) =",format(hasil_1006,"08b"))
```
Operator `<<` menggeser seluruh bit `angka1_1006` ke kiri sebanyak `jumlah_geser_1006` posisi, mengisi posisi kosong di kanan dengan `0`. Efeknya setara dengan mengalikan angka dengan 2 pangkat jumlah geseran (`angka × 2ⁿ`).
Contoh: `5` = `0101`, digeser 1 → `1010` = `10` (yaitu `5 × 2¹`).

10. Pergeseran bit ke kanan (`>>`)
```python
hasil_1006 = angka1_1006 >> jumlah_geser_1006
print("\nBitwise geser kiri (>>)")
print(angka1_1006,">>",jumlah_geser_1006,"=",hasil_1006)
print("Biner hasil =",bin(hasil_1006))
print("Biner hasil (8 bit) =",format(hasil_1006,"08b"))
```
Operator `>>` menggeser bit ke kanan sebanyak `jumlah_geser_1006` posisi. Untuk bilangan **positif**, ini setara dengan pembagian bulat (*floor division*) dengan 2 pangkat jumlah geseran (`angka // 2ⁿ`).
Contoh: `10` = `1010`, digeser 1 → `0101` = `5` (yaitu `10 // 2¹`).

---

**Ringkasan perbedaan operator bitwise:**

| Operator | Nama | Cara kerja |
|---|---|---|
| `&` | AND | 1 jika kedua bit = 1 |
| `\|` | OR | 1 jika salah satu bit = 1 |
| `^` | XOR | 1 jika kedua bit berbeda |
| `~` | NOT | membalik semua bit (komplemen satu) |
| `<<` | Geser kiri | kali 2ⁿ |
| `>>` | Geser kanan | bagi bulat 2ⁿ |


## Tugas Praktikum

**Soal :**

<div align="center">

<img src="SoalTugas3_2611531006.jpeg" width="500"><br>
<em>Gambar 2.2 : Soal Tugas Praktikum</em>
</div>

**Kode Jawaban :**

a. Kode Program tugas3_2611531006
</div>

In [ ]:
# ==========================================
# SISTEM TRANSAKSI TOKO
# Nama : Maulani Almufida Ilmi
# NIM  : _1006
# ==========================================

print("=== SISTEM TRANSAKSI TOKO ===")

# Input pelanggan
nama_1006 = input("Nama Pelanggan : ")
status_1006 = input("Status (member/nonmember) : ")
harga_1006 = float(input("Total Belanja : Rp"))
jumlah_1006 = int(input("Jumlah Barang : "))
promo_1006 = input("Kode Promo : ")

# Data promo
daftar_promo_1006 = ["HEMAT10", "HEMAT20", "GRATISONGKIR"]

# Perbandingan
member_1006 = status_1006 == "member"
belanja_1006 = harga_1006 >= 200000
barang_1006 = jumlah_1006 >= 3

# Membership
promo_ada_1006 = promo_1006 in daftar_promo_1006
promo_tidak_ada_1006 = promo_1006 not in daftar_promo_1006

# Logika
diskon_ok_1006 = member_1006 and belanja_1006
promo_ok_1006 = barang_1006 or promo_ada_1006
bukan_member_1006 = not member_1006

# Aritmatika
diskon_1006 = harga_1006 * 10 / 100 if diskon_ok_1006 else 0
total_1006 = harga_1006 - diskon_1006
rata_1006 = harga_1006 / jumlah_1006
sisa_1006 = jumlah_1006 % 2

# Penugasan
poin_1006 = 0
poin_1006 += int(harga_1006 // 10000)

# Identity
a_1006 = ["member"]
b_1006 = ["member"]

# Bitwise
kode_1006 = 0

if member_1006:
    kode_1006 |= 1

if belanja_1006:
    kode_1006 |= 2

if barang_1006:
    kode_1006 |= 4

if promo_ada_1006:
    kode_1006 |= 8

# AND dan XOR
cek_member_1006 = kode_1006 & 1
cek_promo_1006 = kode_1006 & 8
hasil_xor_1006 = kode_1006 ^ 11

# ==========================================
# HASIL
# ==========================================

print("\n=== DATA PELANGGAN ===")
print("Nama          :", nama_1006)
print("Status        :", status_1006)
print("Total Belanja : Rp", harga_1006)
print("Jumlah Barang :", jumlah_1006)
print("Kode Promo    :", promo_1006)

print("\n=== HASIL VALIDASI ===")
print("Belanja >= 200000 :", belanja_1006)
print("Barang >= 3       :", barang_1006)
print("Member            :", member_1006)
print("Promo tersedia    :", promo_ada_1006)
print("Dapat Diskon      :", diskon_ok_1006)
print("Dapat Promo       :", promo_ok_1006)
print("Bukan Member      :", bukan_member_1006)

print("\n=== HASIL PERHITUNGAN ===")
print("Diskon 10%        : Rp", diskon_1006)
print("Total Bayar       : Rp", total_1006)
print("Rata-rata Barang  : Rp", rata_1006)
print("Sisa Pembagian    :", sisa_1006)
print("Poin              :", poin_1006)

print("\n=== OPERATOR IDENTITY ===")
print("a is b            :", a_1006 is b_1006)
print("a is not b        :", a_1006 is not b_1006)
print("a == b            :", a_1006 == b_1006)

print("\n=== OPERATOR BITWISE ===")
print("Kode Biner        :", format(kode_1006, "04b"))
print("Kode Desimal      :", kode_1006)
print("Cek Member (&)    :", format(cek_member_1006, "04b"))
print("Cek Promo (&)     :", format(cek_promo_1006, "04b"))
print("Hasil XOR (^)     :", format(hasil_xor_1006, "04b"))

In [ ]:

=== HASIL PERHITUNGAN ===
Diskon 10%        : Rp 30000
Total Bayar       : Rp 270000
Rata-rata Barang  : Rp 60000
Sisa Pembagian    : 1
Poin              : 30

=== OPERATOR BITWISE ===
Kode Biner        : 1111
Kode Desimal      : 15
Cek Member (&)    : 0001
Cek Promo (&)     : 1000
Hasil XOR (^)     : 0100

<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

**Penjelasan :** 
| Operator     | Contoh dalam program | Fungsi                                 |
| ------------ | -------------------- | -------------------------------------- |
| Aritmatika   | `*`, `-`             | Menghitung diskon dan pembayaran       |
| Perbandingan | `==`, `>=`           | Membandingkan status dan total belanja |
| Logika       | `and`                | Memastikan dua syarat terpenuhi        |
| Assignment   | `=`, `+=`            | Menyimpan nilai ke variabel            |
| Kondisional  | `if`, `elif`, `else` | Menentukan diskon dan akses promo      |

</div>


### 1. Identitas Program

```python
# SISTEM TRANSAKSI TOKO
# Nama : Maulani Almufida Ilmi
# NIM  : ********1006
```

Bagian ini berisi identitas mahasiswa dan judul program. Semua variabel dalam program menggunakan akhiran **`_1006`** sesuai 4 digit terakhir NIM.

---

### 2. Input Data Pelanggan

```python
nama_1006 = input("Nama Pelanggan : ")
status_1006 = input("Status (member/nonmember) : ")
harga_1006 = float(input("Total Belanja : Rp"))
jumlah_1006 = int(input("Jumlah Barang : "))
promo_1006 = input("Kode Promo : ")
```

Bagian ini digunakan untuk memasukkan data pelanggan.

* `nama_1006` → menyimpan nama pelanggan.
* `status_1006` → menyimpan status member/nonmember.
* `harga_1006` → menyimpan total harga belanja.
* `jumlah_1006` → menyimpan jumlah barang.
* `promo_1006` → menyimpan kode promo.

`float` digunakan untuk harga karena harga dapat memiliki angka desimal, sedangkan `int` digunakan untuk jumlah barang karena jumlah barang berupa bilangan bulat.

---

### 3. Daftar Kode Promo

```python
daftar_promo_1006 = ["HEMAT10", "HEMAT20", "GRATISONGKIR"]
```

Bagian ini membuat daftar kode promo yang tersedia di toko.

Kode yang tersedia adalah:

* HEMAT10
* HEMAT20
* GRATISONGKIR

---

### 4. Operator Perbandingan

```python
member_1006 = status_1006 == "member"
belanja_1006 = harga_1006 >= 200000
barang_1006 = jumlah_1006 >= 3
```

Bagian ini menggunakan operator perbandingan.

* `==` → membandingkan apakah status pelanggan adalah member.
* `>=` → memeriksa apakah belanja minimal Rp200.000.
* `>=` → memeriksa apakah jumlah barang minimal 3.

Hasilnya berupa **True atau False**.

Dengan data:

```text
status = member
harga = 300000
jumlah = 5
```

maka semuanya menghasilkan `True`.

---

### 5. Operator Membership

```python
promo_ada_1006 = promo_1006 in daftar_promo_1006
promo_tidak_ada_1006 = promo_1006 not in daftar_promo_1006
```

Bagian ini menggunakan operator **`in`** dan **`not in`**.

* `in` digunakan untuk mengecek apakah kode promo ada di dalam daftar.
* `not in` digunakan untuk mengecek apakah kode promo tidak ada di dalam daftar.

Karena menggunakan `HEMAT10`, maka:

```text
promo_ada_1006 = True
promo_tidak_ada_1006 = False
```

---

### 6. Operator Logika

```python
diskon_ok_1006 = member_1006 and belanja_1006
promo_ok_1006 = barang_1006 or promo_ada_1006
bukan_member_1006 = not member_1006
```

Bagian ini menggunakan operator **`and`**, **`or`**, dan **`not`**.

**`and`**

```python
member_1006 and belanja_1006
```

Artinya pelanggan mendapatkan diskon jika **member DAN belanja minimal Rp200.000**.

**`or`**

```python
barang_1006 or promo_ada_1006
```

Artinya pelanggan mendapatkan promo jika **jumlah barang memenuhi syarat ATAU kode promo tersedia**.

**`not`**

```python
not member_1006
```

Digunakan untuk mengetahui apakah pelanggan **bukan member**.

---

### 7. Operator Aritmatika

```python
diskon_1006 = harga_1006 * 10 / 100 if diskon_ok_1006 else 0
total_1006 = harga_1006 - diskon_1006
rata_1006 = harga_1006 / jumlah_1006
sisa_1006 = jumlah_1006 % 2
```

Bagian ini digunakan untuk menghitung transaksi.

**Menghitung diskon:**

```python
harga_1006 * 10 / 100
```

Rp300.000 × 10% = **Rp30.000**

**Menghitung total pembayaran:**

```python
harga_1006 - diskon_1006
```

Rp300.000 − Rp30.000 = **Rp270.000**

**Menghitung rata-rata harga:**

```python
harga_1006 / jumlah_1006
```

Rp300.000 ÷ 5 = **Rp60.000**

**Operator `%`:**

```python
jumlah_1006 % 2
```

Digunakan untuk mencari sisa pembagian.

5 ÷ 2 memiliki sisa **1**.

---

### 8. Operator Penugasan

```python
poin_1006 = 0
poin_1006 += int(harga_1006 // 10000)
```

Bagian ini menggunakan operator **`+=`**.

Awalnya:

```text
poin = 0
```

Kemudian:

```text
300000 // 10000 = 30
```

Sehingga:

```text
poin = 0 + 30
```

Hasil akhirnya **30 poin**.

---

### 9. Operator Identity

```python
a_1006 = ["member"]
b_1006 = ["member"]

print(a_1006 is b_1006)
print(a_1006 is not b_1006)
print(a_1006 == b_1006)
```

Bagian ini digunakan untuk membedakan **`is`** dengan **`==`**.

* `is` → membandingkan **identitas objek**.
* `is not` → memeriksa apakah identitas objek berbeda.
* `==` → membandingkan **nilai/isi objek**.

Walaupun `a_1006` dan `b_1006` memiliki isi yang sama, keduanya merupakan objek yang berbeda.

Maka hasilnya:

```text
a is b       : False
a is not b   : True
a == b       : True
```

---

### 10. Operator Bitwise

```python
kode_1006 = 0

if member_1006:
    kode_1006 |= 1

if belanja_1006:
    kode_1006 |= 2

if barang_1006:
    kode_1006 |= 4

if promo_ada_1006:
    kode_1006 |= 8
```

Bagian ini mengubah kondisi pelanggan menjadi kode biner.

Nilainya:

| Nilai  | Kondisi             |
| ------ | ------------------- |
| `0001` | Member              |
| `0010` | Belanja ≥ Rp200.000 |
| `0100` | Barang ≥ 3          |
| `1000` | Promo tersedia      |

Karena semua kondisi terpenuhi:

```text
0001
0010
0100
1000
----
1111
```

Sehingga kode transaksi:

```text
Biner    : 1111
Desimal  : 15
```

---

### 11. Operator AND Bitwise

```python
cek_member_1006 = kode_1006 & 1
cek_promo_1006 = kode_1006 & 8
```

Operator `&` digunakan untuk **memeriksa bit tertentu**.

Untuk member:

```text
1111
0001
----
0001
```

Hasilnya `0001`, berarti pelanggan adalah member.

Untuk promo:

```text
1111
1000
----
1000
```

Hasilnya `1000`, berarti kode promo tersedia.

---

### 12. Operator XOR

```python
hasil_xor_1006 = kode_1006 ^ 11
```

Operator `^` digunakan untuk membandingkan perbedaan bit antara kode transaksi dan kode referensi.

```text
1111
1011
----
0100
```

Hasilnya:

```text
0100
```

atau **4 dalam desimal**.

---

### 13. Bagian Output

```python
print("Nama          :", nama_1006)
print("Status        :", status_1006)
print("Total Belanja : Rp", harga_1006)
```


```text
Nama          : Maulani Almufida Ilmi
Status        : member
Total Belanja : Rp 300000
Jumlah Barang : 5
Kode Promo    : HEMAT10

Diskon        : Rp 30000
Total Bayar   : Rp 270000
Rata-rata     : Rp 60000

Kode Biner    : 1111
Kode Desimal  : 15
```


---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB III<br>KESIMPULAN</h1>

Berdasarkan praktikum yang telah dilakukan, operator `in` dan `not in` digunakan untuk mengecek keberadaan suatu nilai dalam sebuah data. Operator `in` menghasilkan `True` jika nilai ditemukan, sedangkan `not in` menghasilkan `True` jika nilai tidak ditemukan. Selain itu, operator `is` dan `is not` digunakan untuk membandingkan identitas suatu objek. Pada program, `object1` dan `object2` memiliki identitas yang sama karena merujuk pada objek yang sama.

Praktikum ini juga menunjukkan perbedaan antara `is` dan `==`. Operator `is` membandingkan identitas objek, sedangkan `==` membandingkan isi atau nilai objek. Meskipun `object1` dan `object3` memiliki isi yang sama, keduanya merupakan objek yang berbeda karena `object3` dibuat menggunakan `copy()`. Dengan demikian, pemahaman terhadap operator keanggotaan dan identitas sangat penting agar penggunaan operator dalam Python dapat dilakukan dengan tepat sesuai kebutuhan.

</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">DAFTAR PUSTAKA</h1>

1. Python Software Foundation. (2026). Expressions — Membership test operations and identity comparisons. Python Documentation. https://docs.python.org/3/reference/expressions.html memcite
2. Python Software Foundation. (2026). Data Structures — More on Conditions. Python Documentation. https://docs.python.org/3/tutorial/datastructures.html
3. Python Software Foundation. (2026). Built-in Types — Comparisons. Python Documentation. https://docs.python.org/3/library/stdtypes.html

</div>